gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -232,20 +230,19 @@ public void run() {
 
       try {
         if (shouldPublishDataInTask()) {
-
           // If data should be published by the task, publish the data and set the task state to COMMITTED.
           // Task data can only be published after all forks have been closed by closer.close().
           publishTaskData();
           this.taskState.setWorkingState(WorkUnitState.WorkingState.COMMITTED);
```

Comment:
Seems the taskstate should be set to `COMMITTED` here only if the task succeeded. Otherwise the mapper will not fail and we won't notice the task failed.